### PR TITLE
Change imagePullPolicy to `IfNotPresent`

### PR DESCRIPTION
**Pull request by chipzoller**
Signed-off-by: chipzoller <chipzoller@gmail.com>

## What does this PR change?

Changes the imagePullPolicy where previously set to `Always` to now adopt `IfNotPresent` to be consistent across the chart.

## Does this PR rely on any other PRs?

No

## How does this PR impact users? (This is the kind of thing that goes in release notes!)

Only user-facing impact this may have is to improve startup time when Pods are disrupted and, related, to slightly reduce network traffic since the same images are not re-pulled. The other benefit of this change is Kubecost in that we should see, over time, less egress traffic as fewer image pulls occur.

## Links to Issues or tickets this PR addresses or fixes

N/A

## What risks are associated with merging this PR? What is required to fully test this PR?

Risks should be almost none to users. Since image tags are employed throughout the chart, installation and upgrade should result in new tags which would always be pulled.

## How was this PR tested?

Installed a minimal version locally and verified all Pods' imagePullPolicy fields were set to `IfNotPresent`.

## Have you made an update to documentation? If so, please provide the corresponding PR.

N/A

### CODE REVIEW

File: cost-analyzer/templates/_helpers.tpl [hunk 1]
```diff
@@ -902,7 +902,7 @@ Begin Kubecost 2.0 templates
     periodSeconds: {{ .Values.kubecostAggregator.readinessProbe.periodSeconds }}
     failureThreshold: {{ .Values.kubecostAggregator.readinessProbe.failureThreshold }}
   {{- end }}
-  imagePullPolicy: Always
+  imagePullPolicy: IfNotPresent
```
Comment (thomasvn) [inline cost-analyzer/templates/_helpers.tpl]: Should we use this as an opportunity to template the `imagePullPolicy` here on the Aggregator and also the Forecasting image? Overall this looks good! Just want to make sure that if we change the current default behavior the user has the option to override.

Comment (chipzoller) [reply cost-analyzer/templates/_helpers.tpl]: I knew you were going to ask that :) I agree, but I'd like some help with that from someone else if possible.

Comment (thomasvn) [reply cost-analyzer/templates/_helpers.tpl]: Haha sure thing! I'll try to commit those changes onto this PR later today.

Comment (jessegoodier) [reply cost-analyzer/templates/_helpers.tpl]: There's money to be made in an AI specific helm templater :)

**Comment by thomasvn** [comment]
I've made an update so that `imagePullPolicy` is templated for Aggregator, CloudCost, and Forecasting images. Default is `IfNotPresent`

```txt
helm template ./cost-analyzer | grep -i "imagePullPolicy"
        imagePullPolicy: IfNotPresent
          imagePullPolicy: IfNotPresent
          imagePullPolicy: IfNotPresent
          imagePullPolicy: IfNotPresent
          imagePullPolicy: IfNotPresent
          imagePullPolicy: IfNotPresent
          imagePullPolicy: IfNotPresent
          imagePullPolicy: IfNotPresent
          imagePullPolicy: "IfNotPresent"
```

**Review by thomasvn — APPROVED** [review]

**Comment by chipzoller** [comment]
Bless you 🙏

**Comment by AjayTripathy** [comment]
LGTM -- in favor of improved startup time! Opencost does this as well.

**Pull request merged**